feat: 댓글 조회 기능 구현

dain391 opened the pull request:
## 이슈 번호
#17

## 작업 내용
- 댓글 레포지토리에 조회 메서드 추가
- 서비스 계층에서 필터링 및 DTO 변환 처리
- 컨트롤러에서 API 응답 표준화 적용 (ApiResponse)

## 스크린샷(선택)

dain391 committed:
feat: 댓글 조회 기능 구현

- taskId 기준 댓글 조회 API 구현
- 삭제되지 않은 댓글만 최신순 정렬로 반환
- 응답에 CreateCommentResponseDto 재사용

resolves: #17

Jaehyeon-Han commented on src/main/java/com/sparta/taskflow/domain/comment/service/CommentService.java:
```diff
@@ -24,4 +26,11 @@ public CreateCommentResponseDto createComment(CreateCommentRequestDto requestDto
+        List<Comment> comments = commentRepository.findAllByTaskIdAndIsDeletedFalseOrderByCreatedAtDesc(
+            taskId);
+        return comments.stream()
+                       .map(CreateCommentResponseDto::of)
```
조회 응답 DTO라면 CreateCommentResponseDto 이름은 혼동이 올 것 같아요. 원래 생성이랑 수정이랑 같은 데이터 형식이고 표현하는 게 Comment라면 처음에 사용하신대로 CommentResponseDto가 더 나을 것 같아요.

그리고 도전과제로 명세로 넘어오면서 API 페이지랑 달라진 부분이 있어서 수정해주셔야할 것 같습니다.

dain391 committed:
refactor: 댓글 조회 기능 명세서 기준으로 수정

- 조회 URL을 /api/tasks/{taskId}/comments 로 변경
- 페이지네이션(page, size) 파라미터 적용
- 응답 타입을 Page<CommentResponseDto>로 변경
- 유저 정보 포함을 위해 UserResponseDto 적용

resolves: #17

Jaehyeon-Han commented on src/main/java/com/sparta/taskflow/domain/comment/dto/request/CreateCommentRequestDto.java:
```diff
@@ -1,9 +1,11 @@
+import lombok.Setter;
 
 @Getter
+@Setter
```
RequestDto에 Setter는 왜 필요한가요?

Jaehyeon-Han commented on src/main/java/com/sparta/taskflow/domain/comment/service/CommentService.java:
```diff
@@ -26,11 +34,16 @@ public CreateCommentResponseDto createComment(CreateCommentRequestDto requestDto
+        Page<Comment> comments = commentRepository
+            .findAllByTaskIdAndIsDeletedFalseOrderByCreatedAtDesc(taskId, pageable);
+
+        return comments.map(comment -> {
```
N+1 문제 생길 것 같아요.

dain391 committed:
refactor: Setter 제거 및 taskId 직접 주입 방식으로 수정

- CreateCommentRequestDto에서 @Setter 제거
- Controller에서 taskId를 set하는 방식 대신, 서비스에 전달하는 방식으로 변경
- CommentService에 createComment 메서드 오버로딩 대응

resolves: #17

Jaehyeon-Han commented on src/main/java/com/sparta/taskflow/domain/comment/service/CommentService.java:
```diff
@@ -1,36 +1,50 @@
+    private final UserRepository userRepository;
 
-    public CreateCommentResponseDto createComment(CreateCommentRequestDto requestDto) {
+    // 댓글 생성 메서드
```
안전하게 하려면 Task 조회 후에 댓글을 생성해야 할 거예요. 한 사람이 댓글을 다는 도중에 Task를 삭제해버릴 수 있어서요. 하지만 지금 상황에서는 크게 문제되지 않을 것 같아요.

dain391 committed:
refactor: 댓글 DTO 통합 및 단순화

- CreateCommentRequestDto와 CommentRequestDto 통합
- CreateCommentResponseDto와 CommentResponseDto 통합
- 코드 간결화 및 관리 편의성 향상

resolves: #17
